src/lender/assessor.sol

```diff
@@ -164,19 +193,30 @@ contract Assessor is Definitions, Auth, Interest {
         return rdiv(seniorAssetValue, seniorTranche.tokenSupply());
     }
 
-    function calcJuniorTokenPrice(uint epochNAV, uint epochReserve) public view returns(uint) {
-        if ((epochNAV == 0 && epochReserve == 0) || juniorTranche.tokenSupply() == 0) {
+    function _calcJuniorTokenPrice(uint nav_, uint reserve_) internal view returns (uint) {
+        if ((nav_ == 0 && reserve_ == 0) || juniorTranche.tokenSupply() == 0) {
             // initial token price at start 1.00
             return ONE;
         }
-        uint totalAssets = safeAdd(epochNAV, epochReserve);
+        // reserve includes creditline from maker
+        uint totalAssets = safeAdd(nav_, reserve_);
+
+        // includes creditline from mkr
         uint seniorAssetValue = calcExpectedSeniorAsset(seniorDebt(), seniorBalance_);
 
         if(totalAssets < seniorAssetValue) {
             return 0;
         }
 
-        return rdiv(safeSub(totalAssets, seniorAssetValue), juniorTranche.tokenSupply());
+        // the junior tranche only needs to pay for the mkr over-collateralization if
+        // the mkr vault is liquidated, if that is true juniorStake=0
+        uint juniorStake = 0;
```

Comment:
We could make a method for getting the juniorStake together with the modifier which checks if a lender exists otherwise return 0.